itests/fevm_test.go

```diff
@@ -1199,3 +1199,92 @@ func deployContractWithEth(ctx context.Context, t *testing.T, client *kit.TestFu
 		S:                    big.Zero(),
 	}
 }
+
+func TestEthGetTransactionCount(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	// Create a new Ethereum account
+	key, ethAddr, filAddr := client.EVM().NewAccount()
+
+	// Test initial state (should be zero)
+	initialCount, err := client.EVM().EthGetTransactionCount(ctx, ethAddr, ethtypes.NewEthBlockNumberOrHashFromPredefined("latest"))
+	require.NoError(t, err)
+	require.Zero(t, initialCount)
+
+	// Send some funds to the new account (this shouldn't change the nonce)
+	kit.SendFunds(ctx, t, client, filAddr, types.FromFil(10))
+
+	// Check nonce again (should still be zero)
+	count, err := client.EVM().EthGetTransactionCount(ctx, ethAddr, ethtypes.NewEthBlockNumberOrHashFromPredefined("latest"))
+	require.NoError(t, err)
+	require.Zero(t, count)
+
+	// Prepare and send multiple transactions
+	numTx := 5
+	var lastHash ethtypes.EthHash
+
+	contractHex, err := os.ReadFile("./contracts/SelfDestruct.hex")
+	require.NoError(t, err)
+	contract, err := hex.DecodeString(string(contractHex))
+	require.NoError(t, err)
+
+	gasParams, err := json.Marshal(ethtypes.EthEstimateGasParams{Tx: ethtypes.EthCall{
+		From: &ethAddr,
+		Data: contract,
+	}})
+	require.NoError(t, err)
+	gaslimit, err := client.EthEstimateGas(ctx, gasParams)
+	require.NoError(t, err)
+	for i := 0; i < numTx; i++ {
+		tx := &ethtypes.Eth1559TxArgs{
+			ChainID:              buildconstants.Eip155ChainId,
+			To:                   &ethAddr, // sending to self
+			Value:                big.NewInt(1000),
+			Nonce:                i,
+			MaxFeePerGas:         types.NanoFil,
+			MaxPriorityFeePerGas: types.NanoFil,
+			GasLimit:             int(gaslimit),
+		}
+		client.EVM().SignTransaction(tx, key.PrivateKey)
+		lastHash = client.EVM().SubmitTransaction(ctx, tx)
+
+		// Check pending transaction count immediately after submission
+		pendingCount, err := client.EVM().EthGetTransactionCount(ctx, ethAddr, ethtypes.NewEthBlockNumberOrHashFromPredefined("pending"))
+		require.NoError(t, err)
+		require.Equal(t, ethtypes.EthUint64(i+1), pendingCount, "Pending transaction count should increment immediately")
```

Comment:
Sorry, my bad, but I think this is at risk of being flaky - there's a small chance it gets into a block and out of the mpool between submission and GetTransactionCount; so how about instead of an exact check, you do a "between" `i` and `i+1`; either as two asserts, or as one `require.True` with that particular check and `fmt.Sprintf()` the message so we can see what it is.

Reply:
I feel this should work @rvagg 

```go
pendingCount, err := client.EVM().EthGetTransactionCount(ctx, ethAddr, ethtypes.NewEthBlockNumberOrHashFromPredefined("pending"))
require.NoError(t, err)
require.GreaterOrEqual(t, pendingCount, ethtypes.EthUint64(i+1), fmt.Sprintf("Pending transaction count should be at least %d, but got %d", i+1, pendingCount))
```

Reply:
If you're doing a `>=` then you'd want the lower of possible values, which would be `i`, but you should also check for the weird cases where it's bigger than its should be, hence the need for an upper bound.

The lower bound of `i` is needed (I theorise) because you push the message into the mpool, but time has moved on between that action and the `GetTransactionCount`. Likely it's almost always going to remain in the mpool for that next call, but it's possible that a new block was minted between the two calls and it happened to move from the mpool into a new block. It's unlikely, but this is the nature of racy tests that you have to account for, running with `-race` may even pick it up occasionally. The aim should be to remove probability from the equation wherever possible and thereby remove the possibility of flaky tests.

So. `count` is almost always going to be `i+1` but it may also possibly be `i` in some cases.

Optionally just remove this check, it's nice to have in there so we can make sure `pending` doesn't do weird things, but if you feel it's getting too verbose then it's not a huge loss.